Managers/LuaMan.cpp

```diff
@@ -576,17 +576,21 @@ namespace RTE {
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	bool LuaMan::SaveCurrentGame(const std::string &fileName) {
+	int LuaMan::SaveCurrentGame(const std::string &fileName) {
 		Scene *scene = g_SceneMan.GetScene();
-		GAScripted *activity = dynamic_cast<GAScripted*>(g_ActivityMan.GetActivity());
+		GAScripted *activity = dynamic_cast<GAScripted *>(g_ActivityMan.GetActivity());
 
-		if (!scene || !activity) {
-			return false;
+		if (!scene || !activity || (activity && activity->GetActivityState() == Activity::ActivityState::Over)) {
+			return 1;
```

Comment:
Let's use an enum if we can, not int codes

Reply:
I'll chuck in a todo for now and leave the comment open. Maxim's probably gonna move this method to activityman or something, and maybe he'll swap it when he does.
The minor irritant is that it has to be handleable from lua, so it can't be a class enum and I'll need to luabind the enum (I guess).

Reply:
There isn't really an issue with binding an enum class if you cast the values to int when you bind, e.g.
![image](https://user-images.githubusercontent.com/50018358/207929886-019a758c-0f97-4c08-9394-6242a48aa808.png)
Ideally explicitly specifying the underlaying type would work, but it doesn't, so you need this cast.
Also when "binding" an enum value i'm pretty sure you're not actually binding anything, it's just making a copy of the value and it's nothing like actual binding of functions and member variables.

The problem is when you try to call something from Lua that has an enum class parameter, for that you'll need an adapter that accepts int instead and casts it to the enum class.

Reply:
Ahh right. Either way, I didn't wanna bother haha
I figure you can do it if/when you cleanup and move it to activity man :D